compiler/rustc_infer/src/traits/util.rs

```diff
@@ -185,7 +199,19 @@ impl<'tcx> Elaborator<'tcx> {
                 // though conceivably we might.
             }
             ty::PredicateKind::Clause(ty::Clause::Projection(..)) => {
-                // Nothing to elaborate in a projection predicate.
+                // `<T as ~const Trait>::TYPE` implies `<T as Trait>::TYPE`
```

Comment:
can you add a comment that this is only correct if the predicate has exactly 1 effect param? If there were multiple params i could imagine us wanting to add all possible combinations of `const` + `~const`.